libraries/Microsoft.Bot.Builder/Skills/SkillHandlerImpl.cs

```diff
@@ -175,14 +175,27 @@ private async Task<ResourceResponse> ProcessActivityAsync(ClaimsIdentity claimsI
                 {
                     case ActivityTypes.EndOfConversation:
                         await _conversationIdFactory.DeleteConversationReferenceAsync(conversationId, cancellationToken).ConfigureAwait(false);
-                        ApplyEoCToTurnContextActivity(turnContext, activity);
-                        await _bot.OnTurnAsync(turnContext, ct).ConfigureAwait(false);
+                        await SendToBotAsync(activity, turnContext, ct).ConfigureAwait(false);
                         break;
                     case ActivityTypes.Event:
-                        ApplyEventToTurnContextActivity(turnContext, activity);
-                        await _bot.OnTurnAsync(turnContext, ct).ConfigureAwait(false);
+                        await SendToBotAsync(activity, turnContext, ct).ConfigureAwait(false);
```

Comment:
ApplyEventToTurnContextActivity?

Reply:
Thanks eric, that is a mistake, I will fix in a bit

Reply:
Addressed in a new commit, let me know what you think